cg/meta/workflow/nf_analysis.py

```diff
@@ -63,43 +66,103 @@ def get_workflow_manager(self) -> str:
+        """Return a path where the <case>_metrics_deliverables.yaml file should be located."""
+        return Path(self.root_dir, case_id, f"{case_id}_metrics_deliverables").with_suffix(
+            FileExtensions.YAML
+        )
+
+    def get_params_file_path(self, case_id: str, params_file: Optional[Path] = None) -> Path:
+        """Return parameters file or a path where the default parameters file for a case id should be located."""
+        if params_file:
+            return Path(params_file).absolute()
+        return Path((self.get_case_path(case_id)), f"{case_id}_params_file").with_suffix(
+            FileExtensions.YAML
+        )
+
+    def create_case_directory(self, case_id: str, dry_run: bool = False) -> None:
+        """Create case directory."""
+        if not dry_run:
+            os.makedirs(self.get_case_path(case_id=case_id), exist_ok=True)
+
+    def extract_read_files(
+        self, metadata: list, forward: bool = False, reverse: bool = False
+    ) -> List[str]:
+        """Extract a list of fastq file paths for either forward or reverse reads."""
+        if forward and not reverse:
+            read_number = 1
+        elif reverse and not forward:
+            read_number = 2
+        else:
+            raise ValueError("Either forward or reverse needs to be specified")
+        sorted_metadata: list = sorted(metadata, key=operator.itemgetter("path"))
+        return [d["path"] for d in sorted_metadata if d["read"] == read_number]
+
     def verify_case_config_file_exists(self, case_id: str, dry_run: bool = False) -> None:
         """Raise an error if config file is not found."""
         if not dry_run and not Path(self.get_case_config_path(case_id=case_id)).exists():
             raise ValueError(f"No config file found for case {case_id}")
 
-    def get_deliverables_file_path(self, case_id: str) -> Path:
-        """Returns a path where the deliverables file for a case should be located."""
-        return NextflowAnalysisAPI.get_deliverables_file_path(
-            case_id=case_id, root_dir=self.root_dir
-        )
-
     def verify_deliverables_file_exists(self, case_id: str) -> None:
         """Raise an error if deliverables files file is not found."""
         if not Path(self.get_deliverables_file_path(case_id=case_id)).exists():
             raise CgError(f"No deliverables file found for case {case_id}")
 
-    def get_metrics_deliverables_path(self, case_id: str) -> Path:
-        """Return a path where the <case>_metrics_deliverables.yaml file should be located."""
-        return Path(self.root_dir, case_id, f"{case_id}_metrics_deliverables").with_suffix(
-            FileExtensions.YAML
+    def get_replace_map(self, case_id: str) -> dict:
+        """Get a mapping to replace constants from template to create case deliverables."""
+        return {
+            "PATHTOCASE": str(self.get_case_path(case_id)),
+            "CASEID": case_id,
+        }
```

Comment:
Is it possible to use a data model here instead of passing a dictionary?
```suggestion
    def get_replace_map(self, case_id: str) -> dict:
        """Get a mapping to replace constants from template to create case deliverables."""
        return {
            "PATHTOCASE": str(self.get_case_path(case_id)),
            "CASEID": case_id,
        }
```